bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieClientImpl.java

```diff
@@ -179,13 +179,17 @@ public long getNumPendingRequests(BookieSocketAddress address, long ledgerId) {
 
     @Override
     public PerChannelBookieClient create(BookieSocketAddress address, PerChannelBookieClientPool pcbcPool,
-            SecurityHandlerFactory shFactory) throws SecurityException {
+            SecurityHandlerFactory shFactory, int bookieProtocolVersion) throws SecurityException {
         StatsLogger statsLoggerForPCBC = statsLogger;
         if (conf.getLimitStatsLogging()) {
             statsLoggerForPCBC = NullStatsLogger.INSTANCE;
         }
-        return new PerChannelBookieClient(conf, executor, eventLoopGroup, allocator, address, statsLoggerForPCBC,
-                authProviderFactory, registry, pcbcPool, shFactory);
+        ClientConfiguration clientConfiguration = conf;
```

Comment:
Oh, you need to clone the clientConfiguration.

So I would suggest having a variable in BookieClientImpl `v3Conf` then use v3Conf if `ProtocolVersion` is V3.

```
if (v3Conf == null) {
    v3Conf = new ClientConfiguration(conf);
    v3Conf.setUseV2WriteProtocol(false);
}
```

Reply:
Yeah, that my mistake. I shouldn't copy it directly. Thanks, fix it later